tools/SetupFlow/DevHome.SetupFlow.ElevatedComponent/ElevatedComponentOperation.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using DevHome.SetupFlow.Common.Contracts;
+using DevHome.SetupFlow.Common.Helpers;
+using DevHome.SetupFlow.ElevatedComponent.Helpers;
+using DevHome.SetupFlow.ElevatedComponent.Tasks;
+using Windows.Foundation;
+
+namespace DevHome.SetupFlow.ElevatedComponent;
+
+/// <summary>
+/// Class for executing operations in the elevated background process.
+/// </summary>
+public sealed class ElevatedComponentOperation : IElevatedComponentOperation
+{
+    private readonly TasksArguments _tasksArguments;
+
+    public ElevatedComponentOperation(IList<string> tasksArgumentList)
+    {
+        _tasksArguments = TasksArguments.FromArgumentList(tasksArgumentList);
+    }
+
+    public void WriteToStdOut(string value)
+    {
+        Console.WriteLine(value);
+    }
+
+    public IAsyncOperation<ElevatedInstallTaskResult> InstallPackageAsync(string packageId, string catalogName)
+    {
+        var taskArgument = _tasksArguments.InstallPackages?.FirstOrDefault(def => def.PackageId == packageId && def.CatalogName == catalogName);
+        if (taskArgument == null)
+        {
+            Log.Logger?.ReportError(Log.Component.Elevated, $"Failed to install '{packageId}' from '{catalogName}' because it was not found");
```

Comment:
Nit: I'd change the log string. "Not found" doesn't seem right in this case. Maybe "<package> will not be installed as it was not in the pre-approved list"